locale/en/docs/guides/event-loop-timers-and-nexttick.md

```diff
@@ -395,6 +398,56 @@ To get around this, the `'listening'` event is queued in a `nextTick()`
 to allow the script to run to completion. This allows the user to set
 any event handlers they want.
 
+### Deduplication
+
+For the `timers` and `check` phases, there is a single transition 
+between C to JavaScript for multiple immediates and timers. This deduplication
+is a form of optimization, which may produce some unexpected side effects.
+Take this code snippet as an example:
+
+```js
+// dedup.js
+const foo = [1, 2];
+const bar = ['a', 'b'];
+
+foo.forEach(num => {
+  setImmediate(() => {
+    console.log('setImmediate', num);
+    bar.forEach(char => {
+      process.nextTick(() => {
+        console.log('process.nextTick', char);
+      });
+    });
+  });
+});
+```
+```bash
+$ node dedup.js
+setImmediate 1
+setImmediate 2
+process.nextTick a
+process.nextTick b
+process.nextTick a
+process.nextTick b
+```
```

Comment:
You're right. Looks like something changed for node 11:

``` bash
bash-3.2$ pbpaste
 const foo = [1, 2];
 const bar = ['a', 'b'];

 foo.forEach(num => {
   setImmediate(() => {
     console.log('setImmediate', num);
     bar.forEach(char => {
       process.nextTick(() => {
         console.log('process.nextTick', char);
       });
     });
   });
 });
bash-3.2$ nvm use 10.15.0
Now using node v10.15.0 (npm v6.4.1)
bash-3.2$ pbpaste | node
setImmediate 1
setImmediate 2
process.nextTick a
process.nextTick b
process.nextTick a
process.nextTick b
bash-3.2$ nvm use 11.0.0
Now using node v11.0.0 (npm v6.4.1)
bash-3.2$ pbpaste | node
setImmediate 1
process.nextTick a
process.nextTick b
setImmediate 2
process.nextTick a
process.nextTick b
bash-3.2$ nvm use 11.6.0
Now using node v11.6.0 (npm v6.5.0-next.0)
bash-3.2$ pbpaste | node
setImmediate 1
process.nextTick a
process.nextTick b
setImmediate 2
process.nextTick a
process.nextTick b
```

Reply:
It looks like [guides](https://nodejs.org/en/docs/guides/) is not segregated by version. What would be the best way to indicate that this behavior is only applicable for Node 10 and below? 

Also, what changed in node 11?

Reply:
@OmarDarwish：Maybe this should be notified something like: 'before Nodejs v11.X, ……' and 'in Nodejs v11.X or later,……'

Reply:
I'm sorry, I don't quite get the ask. What would you like me to change?

Reply:
@OmarDarwish ：Sorry I removed your comment by mistake……I removed mine first, well……what I want to ask is since diff version of Nodejs will have diff behaviours. Would you mind clearing them one by one such as: 

>In Nodejs 10, some……, however in nodejs 11, ……